packages/esm-patient-chart-app/src/mark-patient-deceased/deceased-form.workspace.tsx

```diff
@@ -0,0 +1,240 @@
+import { Controller, useForm, type SubmitHandler } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { WarningFilled } from '@carbon/react/icons';
+import { EmptyState, type DefaultPatientWorkspaceProps } from '@openmrs/esm-patient-common-lib';
+import { ExtensionSlot, useLayoutType, showSnackbar, ResponsiveWrapper } from '@openmrs/esm-framework';
+import { markPatientDeceased, useCausesOfDeath, usePatientDeceasedStatus } from './deceased.resource';
+import styles from './deceased-form.scss';
+
+const MarkPatientDeceasedForm: React.FC<DefaultPatientWorkspaceProps> = ({ closeWorkspace, patientUuid }) => {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const memoizedPatientUuid = useMemo(() => ({ patientUuid }), [patientUuid]);
+  const { causesOfDeath, isLoading: isLoadingCausesOfDeath } = useCausesOfDeath();
+  const { deathDate, isDead } = usePatientDeceasedStatus(patientUuid);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const filteredCausesOfDeath = useMemo(() => {
+    if (!searchTerm) {
+      return causesOfDeath;
+    }
+
+    return searchTerm
+      ? fuzzy
+          .filter(searchTerm, causesOfDeath, {
+            extract: (causeOfDeathConcept) => causeOfDeathConcept.display,
+          })
+          .sort((r1, r2) => r1.score - r2.score)
+          .map((result) => result.original)
+      : causesOfDeath;
+  }, [searchTerm, causesOfDeath]);
+
+  const schema = z.object({
+    causeOfDeath: z.string(),
+    deathDate: z.date().nullable(),
+  });
+
+  type MarkPatientDeceasedFormSchema = z.infer<typeof schema>;
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    setError,
+  } = useForm<MarkPatientDeceasedFormSchema>({
+    mode: 'onSubmit',
+    resolver: zodResolver(schema),
+    defaultValues: {
+      causeOfDeath: '',
+      deathDate: isDead
+        ? new Date(dayjs(deathDate).year(), dayjs(deathDate).month(), dayjs(deathDate).date() - 1)
+        : new Date(),
+    },
+  });
+
+  const onSubmit: SubmitHandler<MarkPatientDeceasedFormSchema> = useCallback(
+    (data) => {
+      setIsSubmitting(true);
+      const { causeOfDeath, deathDate } = data;
+
+      if (!causeOfDeath || !deathDate) {
+        if (!deathDate) {
+          setError('deathDate', {
+            type: 'manual',
+            message: t('deathDateRequired', 'Please select the date of death'),
+          });
+        }
+
+        if (!causeOfDeath) {
+          setError('causeOfDeath', {
+            type: 'manual',
+            message: t('causeOfDeathIsRequired', 'Please select the cause of death'),
+          });
+        }
+        setIsSubmitting(false);
+        return;
+      }
+
+      markPatientDeceased(deathDate, patientUuid, causeOfDeath)
+        .then(() => {
+          closeWorkspace();
+          window.location.reload();
```

Comment:
@ibacher, @brandones is there a better revalidation mechanism for the patient object? [fetchCurrentPatient](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-api/src/shared-api-objects/current-patient.ts#L22) doesn't use SWR so I assume calling mutate with that key won't work. Triggering a full page reload here ensures that the patient's deceased status gets reflected in the Patient Header.

Reply:
Also, I've elected to not show a success notification here as it wouldn't stay on the screen long enough before the page reloads. Should I show it anyway and maybe set an artificial delay to communicate to the user explicitly that marking the patient as deceased succeeded?

Reply:
Currently not, but that's horrible. We should probably update `usePatient()` to return something similar to the `mutate()` hook (i.e., to force it to reload).